src/main/java/org/commcare/session/RemoteQuerySessionManager.java

```diff
@@ -226,6 +227,20 @@ public void refreshItemSetChoices() {
         }
     }
 
+
+    public void validateUserAnswers() {
+        OrderedHashtable<String, QueryPrompt> userInputDisplays = getNeededUserInputDisplays();
+        EvaluationContext ec = getEvaluationContextWithUserInputInstance();
+        for (Enumeration en = userInputDisplays.keys(); en.hasMoreElements(); ) {
+            String key = (String)en.nextElement();
+            QueryPrompt queryPrompt = userInputDisplays.get(key);
+            QueryPromptValidation validation = queryPrompt.getValidation();
+            if (validation != null && !((Boolean)validation.getXpath().eval(ec))) {
```

Comment:
can the evaluation context be re-scoped for each of these to allow referencing the current prompt value with `.` or `current()`?